esx_service/vmdk_ops.py

```diff
@@ -1670,22 +1683,46 @@ def set_vol_opts(name, tenant_name, options):
 
     return False
 
-def msg_about_signal(signalnum, details="exiting"):
+def wait_ops_in_flight():
+    stop_count = WAIT_STOP_COUNT
+
+    # Wait until the in-flight ops counter goes down to zero
+    while ((opsCounter.value > 0) and (stop_count > 0)):
```

Comment:
I will use event here. However, coupling this event with Counter implementation may not be a good idea. I introduced this Counter class for generic purposes, so I'd rather keep the implementation generic so that it can be used for other purposes.

Reply:
I don't think about is as a counter. It's more of a reverse counting semaphore - when drops to 0, releases the waiters

Reply:
@msterin I've addressed all your comments except this one in the latest commit. I'm working on this one right now - will send a new commit if this change looks better.